### ISSUE TITLE
Expected training times on GPU

### ISSUE THREAD

**Issue by mjdall** [UKPLab/sentence-transformers#847]
Hi, I'm not sure if this is the right place to be posting this as it's quite a general transformers question.

I'm just wondering what sort of train time you'd expect for fine-tuning `stsb-distilbert-base`, pytorch and tensorflow are both detecting GPUs (The compute I'm using has 2 K80s, training using contrastive loss for semantic similarity).
My train time from a databricks CPU compute vs GPU compute is the same.
I'm getting about 1.48 it/s with a batch size of 16, so it's only about 24 sentences/second being processed.
Does this sound about right?
My train times seem very slow right now and I'm wondering what the options are for speeding this up.
I realize this is quite a general question, I just want to first validate the training time looks about right before looking to optimize in other areas.

**Comment by nreimers** [comment]
It appears that you are not training on a GPU. The training times look quite slow.

Check if `torch.cuda.is_available()` returns true in your python script

**Comment by mjdall** [comment]
Yup, that prints True. 
Here's the output of `nvidia-smi` if that's helpful:
![image](https://user-images.githubusercontent.com/41932643/113245783-2091ff00-9314-11eb-8031-12b57a8e154e.png)
And some package versions, let me know if there's anything else that would help:
* transformers==4.4.2
* torch==1.8.0
* sentence-transformers==1.0.3
* tensorflow==2.3.1

**Comment by nreimers** [comment]
So it looks like that your GPU is used.

You should check if K80 has FP16 cores (I don't think it has these). In that case, make sure that use_amp=False when calling the fit method.

Otherwise the sentence lengths are critical for the speed: Transformers have a quadratic run time with sentence lengths. So if you train with quite long sentences, it will significantly take time.

Otherwise  there are not that many performance speed-ups besides getting a better GPU.

**Comment by mjdall** [comment]
I've just tried running a Tesla V100, as far as I can tell has fp32 cores. Do you think it could be a packaging issue?
![image](https://user-images.githubusercontent.com/41932643/113268193-9d80a100-9333-11eb-99ad-5e0bb76c97ed.png)

**Comment by nreimers** [comment]
I would assume that you don't have the right pytorch with the correct cuda / cudnn version installed.

**Comment by mjdall** [comment]
Will test tomorrow, I think you're right

**Comment by mjdall** [comment]
Ok installed torch==1.7.1 and the speed is now 8it/s during training 48it/s during evaluation compared to 1.4it/s and ~7it/s. Thanks for the help
